replica/compaction_group.hh

```diff
@@ -146,8 +151,32 @@ public:
     }
 };
 
-using compaction_group_vector = utils::chunked_vector<std::unique_ptr<compaction_group>>;
+using compaction_group_ptr = std::unique_ptr<compaction_group>;
+using compaction_group_vector = utils::chunked_vector<compaction_group_ptr>;
+typedef list<compaction_group,
+        member_hook<compaction_group, compaction_group::list_hook_t, &compaction_group::list_hook>,
+        constant_time_size<false>> compaction_group_list;
```

Comment:
We prefer `using` to `typedef`.

Why is it needed?

Reply:
old habit.

Reply:
Why is it needed = why is the list needed in the first place.